Update event_dispatcher.rst for consistency

matthieu88160 opened the pull request:
Improve event naming for example consistency with the what appends next.

The event is initially named `foo.action` that break the consistency with the lines 164 and 169 of the document.

matthieu88160 committed:
Update event_dispatcher.rst for consistency

Improve event naming for example consistency with the what appends next.

HeahDude commented on components/event_dispatcher.rst:
```diff
@@ -135,7 +135,7 @@ A call to the dispatcher's ``addListener()`` method associates any valid
 
     $listener = new AcmeListener();
-    $dispatcher->addListener('acme.action', array($listener, 'onFooAction'));
+    $dispatcher->addListener('foo.action', array($listener, 'onFooAction'));
```
What about renaming all of them 'acme.foo.action'?

matthieu88160 replied:
Due to naming convention described in the previous line of the doc I don't think that naming the event `'acme.foo.action'` is the better choice.
Maybe using `'acme_foo.action'` can be a prefered approach in regard of these naming convention :
* __"Prefix names with a namespace followed by a dot"__
* __"End names with a verb"__

HeahDude replied:
I guess here `foo` is a placeholder for `acme` and `action` a placeholder for a verb, so maybe `acme.foo` would be less confusing?

matthieu88160 replied:
According to the context I agree.

HeahDude replied:
By the way the example I given is perfectly valid, see https://github.com/FriendsOfSymfony/FOSUserBundle/blob/master/FOSUserEvents.php#L26 as an example :)

matthieu88160 replied:
FOS change password action have some sub-event possibilities, it makes more sense to nest these as you point here than the example show into the event dispatcher one.

To use `'acme.foo.action'` I consider using a pattern like `'{vendor}.{context}.{action}'`. Do you agree with this before I update?

HeahDude replied:
I do :)

matthieu88160 committed:
Rename event_dispatcher.rst events

According to comment about the pull request #7547, renaming the events acme.action to acme.foo.action

HeahDude reviewed (approved):
👍

javiereguiluz commented:
I'm sorry but I can't see why this change is needed. In the original article it was called `acme.foo.action`? I can't find it.

matthieu88160 commented:
In the current version the article is not consistent regarding the name of the event between the first one and the last one used in the "Connecting Listeners" section.

First called `acme.action` it is renamed to `foo.action` into and after the comment.

About renaming it to `acme.foo.action`, take a look in the comments of the commit #7547 where @HeahDude offer to use a pattern as __"{vendor}.{context}.{action}"__.

javiereguiluz commented:
@matthieu88160 now I see it!! I don't know why I didn't see it before 🙈 !!  Thanks for fixing this.

matthieu88160 commented:
You'r welcome

xabbuh commented:
Good catch, thanks @matthieu88160.
